opentelemetry-appender-tracing/examples/basic.rs

```diff
@@ -0,0 +1,25 @@
+//! run with `$ cargo run --example basic --all-features
+
+use opentelemetry_appender_tracing::{layer};
+use tracing::{info, error};
+use opentelemetry_sdk::{logs::{LoggerProvider, Config}, Resource};
+use opentelemetry_api::KeyValue;
+use tracing_subscriber::prelude::*;
+
+fn main() {
+    let exporter  = opentelemetry_stdout::LogExporter::default();
+    let provider: LoggerProvider = LoggerProvider::builder()
+    .with_config(
+        Config::default().with_resource(Resource::new(vec![KeyValue::new(
+            "service.name",
+            "log-appender-tracing-example",
+        )])),
+    )
+    .with_simple_exporter(exporter)
+    .build();
+    let layer  = layer::OpenTelemetryTracingBridge::new(&provider);
+    tracing_subscriber::registry().with(layer).init();
+
+    info!(fruit = "apple", price = 2.99, message = "hello from from price");
```

Comment:
[Aaron] qn. Why use tracing for logging at all?
log, slog, tracing (events  API) - 3 populate logs.

tracing-otel already converts events into SpanEvents, so why is that no sufficient already, and why we need another "bridge" to convert to Logs.?
[Cijo]
1. SpanEvents require Span, and Span's sampling affects SpanEvents.
2. OTel has limits/throttle on the number of SpanEvents per Span
3. SpanEvents are given to Exporters, only when the Span itself is ended/dropped, as opposed to Logs(events) being sent to Exporters immediately (depending on Export Processor.)

[Lalit]
tracing allows events to be created outside Span, but SpanEvents in OTel API does not allow it outside tracing API.

The log facade could be used for pure logging., but does not allow structured logs.

Tracing crate already has ability to emit Logs Always (need to explore - https://docs.rs/tracing/latest/tracing/#emitting-log-records)

event!(Level::TRACE, greeting = ?my_struct);
How do we user decide if this should go as span_event or log?
1. We need to investigate/explore this more. (there may be "Filter" that can be leverages Metadata)
2. We need end-user feedback.


Python
Log API
Otel Tracing API (Spans, SpanEvents)

Rust
Log Crate -- the answer is similar to Python/other languages
Tracing crate - that has "events" *and* "Span", so the distinction is not easy to make.
Otel Tracing API (Spans, SpanEvents)

Reply:
> Tracing crate already has ability to emit Logs Always (need to explore - https://docs.rs/tracing/latest/tracing/#emitting-log-records)

You shouldn't enable this feature for `opentelemetry-rust`. it's meant for libraries like hyper or tokio to smoothly transition to `tracing` in a minor semver release, allowing users to not immediately lose logging when then update their dependencies.

> How do we user decide if this should go as span_event or log? [...] We need to investigate/explore this more. (there may be "Filter" that can be leverages Metadata)

Using [`dynamic_filter_fn`](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/fn.dynamic_filter_fn.html) is an option, but be sure to benchmark this since it might be a little too costly. We can always improve it on the tracing side of things though, especially since there is some low-hanging, performance-related fruit in `tracing-subscriber` that we've been meaning to improve. Y'all saying that it's a little too slow would sufficient for me to justify the positive impact of me working on fixing it.